pkg/storegateway/batch_series_test.go

```diff
@@ -808,6 +811,233 @@ func TestBlockSeriesChunkRefsSetsIterator_ErrorPropagation(t *testing.T) {
 	assert.ErrorContains(t, iterator.Err(), "test limit exceeded")
 }
 
+func TestLoadingBatchSet(t *testing.T) {
+	type testBlock struct {
+		ulid   ulid.ULID
+		series []seriesEntry
+	}
+
+	block1 := testBlock{
+		ulid:   ulid.MustNew(1, nil),
+		series: generateSeriesChunks(t, 10),
+	}
+
+	block2 := testBlock{
+		ulid:   ulid.MustNew(2, nil),
+		series: generateSeriesChunks(t, 10),
+	}
+
+	toSeriesChunkRefs := func(block testBlock, seriesIndex int) seriesChunkRefs {
+		series := block.series[seriesIndex]
+
+		chunkRefs := make([]seriesChunkRef, len(series.chks))
+		for i, c := range series.chks {
+			chunkRefs[i] = seriesChunkRef{
+				blockID: block.ulid,
+				ref:     series.refs[i],
+				minTime: c.MinTime,
+				maxTime: c.MaxTime,
+			}
+		}
+
+		return seriesChunkRefs{
+			lset:   series.lset,
+			chunks: chunkRefs,
+		}
+	}
+
+	testCases := map[string]struct {
```

Comment:
These test cases are great! I think there's a main missing case: a series with 2 chunks, each chunk belonging to a different block.